srtcore/core.cpp

```diff
@@ -3105,6 +3240,385 @@ bool CUDT::checkApplyFilterConfig(const std::string &confstr)
+    }
+
+    steady_clock::time_point rcv_buffer_time_base;
+    bool rcv_buffer_wrap_period = false;
+    steady_clock::duration rcv_buffer_udrift;
+    if (m_bTsbPd && gp->getBufferTimeBase(this, (rcv_buffer_time_base), (rcv_buffer_wrap_period), (rcv_buffer_udrift)))
+    {
+        // We have at least one socket in the group, each socket should have
+        // the value of the timebase set exactly THE SAME.
+
+        // In case when we have the following situation:
+
+        // - the existing link is before [LAST30] (so wrap period is off)
+        // - the new link gets the timestamp from [LAST30] range
+        // --> this will be recognized as entering the wrap period, next
+        //     timebase will get added a segment to this value
+        //
+        // The only dangerous situations could be when one link gets
+        // timestamps from the [FOLLOWING30] and the other in [FIRST30],
+        // but between them there's a 30s distance, considered large enough
+        // time to not fill a network window.
+        enterCS(m_RecvLock);
+        m_pRcvBuffer->applyGroupTime(rcv_buffer_time_base, rcv_buffer_wrap_period, m_iTsbPdDelay_ms * 1000, rcv_buffer_udrift);
+        leaveCS(m_RecvLock);
+
+        HLOGF(mglog.Debug,  "AFTER HS: Set Rcv TsbPd mode: delay=%u.%03us GROUP TIME BASE: %s%s",
+                m_iTsbPdDelay_ms/1000,
+                m_iTsbPdDelay_ms%1000,
+                FormatTime(rcv_buffer_time_base).c_str(),
+                rcv_buffer_wrap_period ? " (WRAP PERIOD)" : " (NOT WRAP PERIOD)");
+    }
+    else
+    {
+        HLOGC(mglog.Debug, log << "AFTER HS: (GROUP, but " << (m_bTsbPd ? "FIRST SOCKET is initialized normally)" : "no TSBPD set)"));
+        updateSrtRcvSettings();
+    }
+
+    // This function currently does nothing, just left for consistency
+    // with updateAfterSrtHandshake().
+    updateSrtSndSettings();
+
+    // These are the values that are normally set initially by setters.
+    int32_t snd_isn = m_iSndLastAck, rcv_isn = m_iRcvLastAck;
+    if (!gp->applyGroupSequences(m_SocketID, (snd_isn), (rcv_isn)))
+    {
+        HLOGC(mglog.Debug, log << "synchronizeWithGroup: @" << m_SocketID
+                << " DERIVED ISN: RCV=%" << m_iRcvLastAck << " -> %" << rcv_isn
+                << " (shift by " << CSeqNo::seqcmp(rcv_isn, m_iRcvLastAck)
+                << ") SND=%" << m_iSndLastAck << " -> %" << snd_isn
+                << " (shift by " << CSeqNo::seqcmp(snd_isn, m_iSndLastAck) << ")");
+        setInitialRcvSeq(rcv_isn);
+        setInitialSndSeq(snd_isn);
+    }
+    else
+    {
+        HLOGC(mglog.Debug, log << "synchronizeWithGroup: @" << m_SocketID
+                << " DEFINED ISN: RCV=%" << m_iRcvLastAck
+                << " SND=%" << m_iSndLastAck);
+    }
+}
+
+// [[using locked(this->m_GroupLock)]];
+bool CUDTGroup::getBufferTimeBase(CUDT* forthesakeof, steady_clock::time_point& w_tb, bool& w_wp, steady_clock::duration& w_dr)
```

Comment:
`forthesakeof` is really not the best name for this.

Reply:
This is a socket core, for which the synchronization is being made. Data prepared and extracted by this function will be then used to set on this one entity.